internal/json/types.go

```diff
@@ -369,6 +369,11 @@ func (e Encoder) AppendInterface(dst []byte, i interface{}) []byte {
 	return append(dst, marshaled...)
 }
 
+// AppendType appends the parameter type (as a string) to the input byte slice.
+func (e Encoder) AppendType(dst []byte, i interface{}) []byte {
+	return e.AppendString(dst, fmt.Sprintf("%T", i))
```

Comment:
Please use reflect.TypeOf(i).String() directly instead.

Reply:
Good suggestion. Had to explicitly handle nil but otherwise pretty straightforward.